susemanager-utils/susemanager-sls/src/modules/mgrnet.py

```diff
@@ -0,0 +1,137 @@
+"""
+Module for gathering DNS FQDNs
+"""
+
+import logging
+import re
+import time
+from threading import Thread
+from queue import Queue
+
+import salt.utils.network
+
+log = logging.getLogger(__name__)
+
+
+def __virtual__():
+    """
+    Only works on POSIX-like systems having 'host' or 'nslookup' available
+    """
+    if not (__utils__["path.which"]("host") or __utils__["path.which"]("nslookup")):
+        log.debug("Neither 'host' nor 'nslookup' is available on the system")
+    return True
```

Comment:
It was made to prevent fails for calling `mgrnet.dns_fqdns` if neither `host` nor `nslookup` is available on the system.

Reply:
I don't see the use of loading this module if it then doesn't do anything. What's wrong with returning `False`, as one would expect from Salt modules?

Reply:
Here we have 2 possible options:
1. return `False`  with `virtual` in case if `host` and `nslookup` are absent on the minion. The possible drawbacks to do it this way:
  - need to add error handling in case if `mgrnet.dns_fqdns` is not available in the `__salt__` with the state
  - in case if `host` and `nslookup` become available it will take some time to reload the modules list, probably need to restart the `salt-minion` service to make the module available after installing `bind-utils` package
2. keep as it is now and just merge the values returned by `mgrnet.dns_fqdns` with the values from `network.fqdns`.

I would prefer to keep as it is now to prevent extra complexity with the state using this function.
@nadvornik what do you think? @agraul is correct here that returning `False` is the native solution for salt modules, but not sure if you are ok to modify the module this way.

Reply:
The check for `mgrnet.dns_fqdns` not available is already there:
https://github.com/uyuni-project/uyuni/pull/5770/files#diff-1a0526ffc994048d3015ad1157ef8a0ee2e7b901d4cd6ba6fc950e2e593041d1R134

Reply:
@nadvornik I'm not really sure about the case when the `host` or `nslookup` is available on the system and in case if we return False with virtual function how long it could take for the module to become available after installing the tools, I'll try to check on the test system. Do you have any changes to the way how to make `bind-utils` installed on the system? State or system installation profile for example.

Reply:
For retail use case we need it for detecting FQDNs of a branch server. `bind-utils` is always installed there.
For other cases it might be enough to say in the documentation that the FQDN detection can be improved by installing `bind-utils` and running HW detection again.